tests/sentry/incidents/test_models.py

```diff
@@ -553,3 +554,44 @@ def test_update(self):
         assert AlertRuleActivity.objects.filter(
             alert_rule=self.alert_rule, type=AlertRuleActivityType.UPDATED.value
         ).exists()
+
+
+class AlertRuleTest(TestCase):
+    def test_processor_hooks_default(self):
+        alert_rule = self.create_alert_rule()
+        assert alert_rule._processor_hooks == []
+
+    def test_processor_hooks_activated(self):
+        alert_rule = self.create_alert_rule(monitor_type=AlertRuleMonitorType.ACTIVATED)
+        subscription = alert_rule.snuba_query.subscriptions.get()
+
+        # Processor hook setup for AlertRuleMonitorType.ACTIVATED
+        assert len(alert_rule._processor_hooks) == 1
+
+        # Add a mock processor hook
+        alert_rule._processor_hooks = [Mock()]
+        alert_rule.process_hooks(subscription)
+
+        assert alert_rule._processor_hooks[0].call_count == 1
+        assert alert_rule._processor_hooks[0].call_args == mock.call(subscription)
+
+    def test_clean_activated_alert_active(self):
```

Comment:
hmmm rather than `activate`/`deactivate`, maybe we can call them `current`/`expired`?